ion_schema_2_0/schema/ion_schema_version_markers.isl

```diff
@@ -47,17 +38,19 @@ $test::{
   invalid_schemas:[
     (
       $ion_schema_0_1
+      schema_header::{}
```

Comment:
🗺️ The rest of the changes in this file are only tangentially related. The `footer` annotations here were incorrect, and may cause these test cases to pass even if the implementation can't detect the bad version marker. I'm also adding `schema_header`s so that we make sure that the version marker is tested independently of the optionality of headers and footers.